Create an OutputStream from a Device and SupportedStreamConfig

DivineGod opened the pull request:
This change adds support for getting an `OutputStream` and `OutputStreamHandle` from a specific device with a specific configuration instead of relying on getting the default configuration.

The new function `try_from_device_config` acts the same as `try_from_device` but adds an extra `spal::SupportedStreamConfig` argument. The caller is then responsible for getting a valid configuration from cpal to pass along.

est31 reviewed (changes requested):
Can you make `try_new_output_stream` call `try_new_output_stream_config`?

DivineGod commented on src/stream.rs:
```diff
@@ -38,6 +38,20 @@ impl OutputStream {
         Ok((out, handle))
     }
 
+    /// Returns a new stream & handle using the given device and stream config.
```
Done

DivineGod commented:
> Can you make `try_new_output_stream` call `try_new_output_stream_config`?

I've refactored the `try_from_device` function to call the `try_from_device_config` function with the default config. I also removed the method `try_new_output_stream` and replaced it with `try_new_output_stream_config` instead.

est31 reviewed (approved):
Thanks!
